### ISSUE TITLE
New Calendar Data model usage from 0.5.0

### ISSUE THREAD

**Issue by tanker11** [amaximus/fkf-garbage-collection#23]
Hello,

I really like your development and used it without issue until now.
With the help of Custom Button Card and templating, I created some indication/warning cards on my Dashboard, but this change resulted in my own card representation does not work any more...
To make it work again, I would need to understand the new concept of the data model, but from the entity states viewer I could not find out what to expect.

I can share the code of the card if that helps in understanding my problem.

Thanks in advance

**Comment by tanker11** [comment]
Hello,

I am not sure how to use them, still, sorry.
My intention is to see the upcoming events like this:
![image](https://user-images.githubusercontent.com/32165691/202680088-d1fb9b78-32c4-48f9-9628-6c78f912eaad.png)
If the date is closer, the icon starts blinking (pulsing).t

Would it be possible to transform it under the new data model?
Ot I should assume it is a calendar entry? Is it possible then to "push" them into a Google Calendar and see the upcoming events there?



Here is the code of the above card with custom button-card:

`type: custom:button-card
entity: sensor.fkf_garbage_collection
show_state: true
show_label: false
name: |
  [[[
     if (states['sensor.fkf_garbage_collection'].attributes.in0 == 0)
    return 'Mai nap';
    return states['sensor.fkf_garbage_collection'].attributes.in0+" nap múlva";
  ]]]
icon: |
  [[[
    
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='green') return 'mdi:leaf';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective') return 'mdi:delete';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='communal') return 'mdi:delete';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective_green') return 'mdi:leaf-circle-outline';
          return "var(--paper-item-icon-color)";   
  ]]]
state_display: |
  [[[
    var var_text="";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='green') var_text="zöldhulladék";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective') var_text="szelektív";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='communal') var_text="kommunális";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective_green') var_text="zöld+szelektív";
    return var_text;
  ]]]
label: |
  [[[
    return states['sensor.fkf_garbage_collection'].attributes.date0;
  ]]]
color_type: icon
styles:
  icon:
    - color: |
        [[[
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='green') return 'green';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective') return 'cyan';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='communal') return 'green';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0=='selective_green') return 'orangered';;
          return "var(--paper-item-icon-color)";
        ]]]
    - animation: |
        [[[
             if (states['sensor.fkf_garbage_collection'].attributes.in0 == 0)
            return "blink 2s ease infinite";
          return "";
        ]]]
`



Thanks in advance.

**Comment by amaximus** [comment]
You should format your code with triple ` instead of single one.

**Comment by amaximus** [comment]
You had all kind of syntax errors.
Here is one working example, however you may need too play around with the colors, etc.:
```
type: custom:button-card
entity: sensor.fkf_garbage_collection
show_state: true
show_label: false
name: |
  [[[
    if (states['sensor.fkf_garbage_collection'].attributes.in0 == 0)
      return 'Mai nap';
    return states['sensor.fkf_garbage_collection'].attributes.in0+" nap múlva";
  ]]]
icon: |
  [[[
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'green')
       return 'mdi:leaf';
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective')
       return 'mdi:delete';
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'communal')
       return 'mdi:delete';
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective_green')
       return 'mdi:leaf-circle-outline';
  ]]]
state_display: |
  [[[
    var var_text="";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'green')
       var_text="zöldhulladék";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective')
       var_text="szelektív";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'communal')
       var_text="kommunális";
    if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective_green')
       var_text="zöld+szelektív";
    return var_text;
  ]]]
color_type: icon
styles:
  icon:
    - color: |
        [[[
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'green')
             return 'green';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective')
             return 'cyan';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'communal')
             return 'green';
          if (states['sensor.fkf_garbage_collection'].attributes.garbage0 == 'selective_green')
             return 'orangered';;
          return "var(--paper-item-icon-color)";
        ]]]
extra_styles: |
  [[[ return `
    @keyframes pulse {
      20% {
        color: ${variables.color};
      }
      80% {
        color: "#e5e5ea";
      }
    }
  `]]]
variables:
  color: var(--paper-item-icon-active-color)
state:
  - value: '0'
    styles:
      icon:
        - animation: pulse 1s infinite
```

**Comment by tanker11** [comment]
Hello,

Many thanks for the code cleaning even if out of scope, it is very useful!

My original issue was actually different: when I upgraded to 0.5.0, the
card I created showed an error.
Then I checked the related entity in the Developer tools, and its content
was totally different from the one I built my code on.
Because your code uses the entity and its attributes in the same way, I
guess it was caused by something else (and it is not persistent any more).

About the citation of the code: it was GitHub suggesting how to embed when
I pressed "Embed code" button... anyway, I also learned from it. :)

Atticus Maximus ***@***.***> ezt írta (időpont: 2022. nov.
19., Szo, 12:35):
